daemon/hooks/hook_logging.go

```diff
@@ -0,0 +1,105 @@
+package hooks
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"sync"
+
+	"github.com/zrepl/zrepl/logger"
+)
+
+type contextKey int
+
+const (
+	contextKeyLog contextKey = 0
+)
+
+type Logger = logger.Logger
+
+func WithLogger(ctx context.Context, log Logger) context.Context {
+	return context.WithValue(ctx, contextKeyLog, log)
+}
+
+func getLogger(ctx context.Context) Logger {
+	if log, ok := ctx.Value(contextKeyLog).(Logger); ok {
+		return log
+	}
+	return logger.NewNullLogger()
+}
+
+const MAX_HOOK_LOG_SIZE int = 1 << 20
```

Comment:
please use envconst

 https://github.com/zrepl/zrepl/blob/c87759affef880d18155a4776dd90a1183912917/util/envconst/envconst.go#L28 

If you need an int, adjust envconst as needed (copy pasta)